### PR TITLE
95 gui hierarchy

### LINKED ISSUE #95: Rewrite hierarchy of GUI classes

### PR DESCRIPTION
**Pull request by mpsijm**
Will close #95 

Biggest. Overhaul. Ever.

I know that pull requests this big shouldn't exist, but I couldn't help myself, I'm sorry!
All `*Screen` classes have been renamed to `*Scene` and all inherit from `AbstractScene`. 
Okay, I'm not gonna type it all out, here's a CRC class diagram:
[Gyazo](https://gyazo.com/b38eed1c97bd1e96787fcaacd3831b23)

Good luck reviewing everyone! :)

<organization,naming,documentation,review

### CODE REVIEW

File: src/main/java/nl/tudelft/ti2206/group9/ShaftEscape.java [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+ * Starting point of the Application.
+ * Contains method to switch Scenes or show Popups.
+ * @author Maarten
+ */
+@SuppressWarnings("restriction")
+public class ShaftEscape extends Application {
+
+	private static Stage stage;
+	/** Width of the Window. */
+	public static final int WIDTH = 480 - 16;
```
Comment (thervh70) [inline src/main/java/nl/tudelft/ti2206/group9/ShaftEscape.java]: Why `-16`? @mpsijm 

< review

File: src/main/java/nl/tudelft/ti2206/group9/ShaftEscape.java [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+	public static final Object TICKER_LOCK = new Object();
+
+	/**
+	 * Start the application in the SplashScene.
+	 * @param appStage the primary Stage for the Application. This is where
+	 * the scenes are shown in.
+	 */
+	@Override
+	public void start(Stage appStage) {
+		stage = appStage;
```
Comment (thervh70) [inline src/main/java/nl/tudelft/ti2206/group9/ShaftEscape.java]: *Auto-generated comment*

    Write to static field nl.tudelft.ti2206.group9.ShaftEscape.stage from instance method nl.tudelft.ti2206.group9.ShaftEscape.start(Stage)

File: src/main/java/nl/tudelft/ti2206/group9/gui/MainMenuScene.java [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+		setButtonFunction(exitButton, BType.EXIT);
+		setButtonFunction(startButton, BType.START);
+		setButtonFunction(settingsButton, BType.SETTINGS);
+
+		return new Node[]{startButton, settingsButton, exitButton};
+	}
+
+	/**
+	 * This method sets the function of a button.
+	 * @param stage given PrimaryStage.
```
Comment (thervh70) [inline src/main/java/nl/tudelft/ti2206/group9/gui/MainMenuScene.java]: This should be removed.

< documentation

**Comment by MKerkhof** [comment]
Review done, if you fix the above things mentioned by Robin, it can be merged!

<review, documentation

File: src/main/java/nl/tudelft/ti2206/group9/ShaftEscape.java (hunk 1 above)
Comment (mpsijm) [inline src/main/java/nl/tudelft/ti2206/group9/ShaftEscape.java]: You should try it yourself, but apparently without the `-16`, the background is too small for the window. There should be a better way to fix this, but this also does the trick.
<review

**Comment by mdingjan** [comment]
As you said, this is indeed a huge PR :P But it's nice to have a hierarchy for the gui now. :)
- I couldn't find anymore checkstyle warnings
- Javadoc is (still) sufficient
- And the tests still cover the most important parts of the gui.
So I'm happy :) Will merge!

<review

**Comment by MathiasMeuleman** [comment]
Looks fine now! Nice job :)

Will merge

<review

**Pull request merged**